### PR TITLE
New version PowerShell 7.4.0-preview.4

### PR DESCRIPTION
**Pull request by anamnavi**
New Version of PowerShell 7.4.0-preview.4

The msix for arm is a temporary solution until we create msi's for it, but we still want to support arm in the meantime.

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/winget-pkgs/pull/110985&drop=dogfoodAlpha

### CODE REVIEW

**Comment by BrandonWanHuanSheng** [comment]
Manifest Error: Inconsistent value in the manifest. [PackageVersion] Value: 7.4.4.0

**Review by ItzLevvie — APPROVED** [review]

**Comment by BrandonWanHuanSheng** [comment]
Blocking Issue: https://github.com/microsoft/winget-cli/issues/163. Because it might require Microsoft.UI.Xaml.2.8

**Comment by mdanish-kh** [comment]
[Policy] Needs-Attention

@stephengillie - Reminder that these PRs require a manual merge as the DisplayVersion is different between MSIX and MSI and we are using the MSI DisplayVersion as PackageVersion since that's the one used for marketing

File: manifests/m/Microsoft/PowerShell/Preview/7.4.0.4/Microsoft.PowerShell.Preview.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- interactive
+- silent
+- silentWithProgress
+UpgradeBehavior: install
+Commands:
+- pwsh
+ReleaseDate: 2023-06-29
+Installers:
+- Architecture: x64
+  InstallerType: wix
```
Comment (CoolPlayLin) [inline manifests/m/Microsoft/PowerShell/Preview/7.4.0.4/Microsoft.PowerShell.Preview.installer.yaml:18]:
```suggestion
  InstallerType: msi
```

File: manifests/m/Microsoft/PowerShell/Preview/7.4.0.4/Microsoft.PowerShell.Preview.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  InstallerUrl: https://github.com/PowerShell/PowerShell/releases/download/v7.4.0-preview.4/PowerShell-7.4.0-preview.4-win-x64.msi
+  InstallerSha256: 30D133F914870645FB4FE84F1C0FC9E7CF5789EE70C3070872995288F252063A
+  ProductCode: '{890ABFDC-3A95-4D6D-ABDE-D4AB5E82768D}'
+- Architecture: x86
+  InstallerType: wix
```
Comment (CoolPlayLin) [inline manifests/m/Microsoft/PowerShell/Preview/7.4.0.4/Microsoft.PowerShell.Preview.installer.yaml:23]:
```suggestion
  InstallerType: msi
```

**Review by CoolPlayLin — CHANGES_REQUESTED** [review]

File: manifests/m/Microsoft/PowerShell/Preview/7.4.0.4/Microsoft.PowerShell.Preview.installer.yaml (hunk 1 above)
Comment (mdanish-kh) [reply manifests/m/Microsoft/PowerShell/Preview/7.4.0.4/Microsoft.PowerShell.Preview.installer.yaml:18]: The MSI is made using WiX Toolset, so `InstallerType: wix` is more accurate in this case. When it comes to passing the silent switches, winget handles both the same way. See

https://github.com/microsoft/winget-cli/blob/e4b9205cff88701b2ad1078dd8ed311ac716b8b6/src/AppInstallerCommonCore/Manifest/ManifestCommon.cpp#L768-L776

**Review by mdanish-kh — APPROVED** [review]

File: manifests/m/Microsoft/PowerShell/Preview/7.4.0.4/Microsoft.PowerShell.Preview.installer.yaml (hunk 1 above)
Comment (CoolPlayLin) [reply manifests/m/Microsoft/PowerShell/Preview/7.4.0.4/Microsoft.PowerShell.Preview.installer.yaml:18]: so what's the issue with this PR?

**Comment by stephengillie** [comment]
> [Policy] Needs-Attention
> 
> @stephengillie - Reminder that these PRs require a manual merge as the DisplayVersion is different between MSIX and MSI and we are using the MSI DisplayVersion as PackageVersion since that's the one used for marketing

Thanks again. Could we use a label and GitHub action to apply it, as a way to automate reminding me of these?

**Pull request merged**